tests/cpp/include/test_core_op.h

```diff
@@ -209,6 +209,13 @@ class CoreOpExecutor : public test::op::OperatorDataInitializer<DType>
           requested.emplace_back(r);
         } else if (req.type == ResourceRequest::kRandom) {
           requested.emplace_back(ResourceManager::Get()->Request(ctx->run_ctx.ctx, req));
+        } else if (req.type == ResourceRequest::kParallelRandom) {
+          Resource rm = ResourceManager::Get()->Request(ctx->run_ctx.ctx, req);
+          if (ctx->run_ctx.ctx.dev_mask() == Context::kCPU) {
+            common::random::RandGenerator<cpu, DType>::AllocState(
```

Comment:
Is there a possibility to define a pre-set seed?

Reply:
I don't know. I got the code from Chris.

Reply:
@cjolivier01 could you help comment for pre-set seed?

Reply:
you define the seed just like you’d define it in the normal code — there’s a C API for it

Reply:
Push